src/utils/safe-compare.ts

```diff
@@ -16,11 +16,14 @@ export default function safeCompare(
     let buffA: Buffer;
     let buffB: Buffer;
 
-    if (typeof strA === 'object' && typeof strB === 'object') {
+    if (typeof strA === 'object') {
```

Comment:
Since we're inside an `if (typeof strA === typeof strB)`, then we inherently know that the `typeof strB` is also an `object` at this point, hence we can revert to the original code and just remove the `&& typeof strB === 'object'` from this `if` statement.